ec/ecresource/deploymentresource/elasticsearch/v2/schema.go

```diff
@@ -483,12 +484,17 @@ func elasticsearchTopologySchema(options topologySchemaOptions) schema.Attribute
 		nodeRolesPlanModifiers = append(nodeRolesPlanModifiers, SetUnknownOnTopologySizeChange())
 	}
 
+	var topologyPlanModifiers []planmodifier.Object
+	if options.tierName == "master" {
+		topologyPlanModifiers = append(topologyPlanModifiers, objectplanmodifier.UseStateForUnknown())
+	}
+
 	return schema.SingleNestedAttribute{
-		Optional: !options.required,
-		// it should be Computed but Computed triggers TF weird behaviour that leads to unempty plan for zero change config
-		// Computed:    true,
-		Required:    options.required,
-		Description: fmt.Sprintf("'%s' topology element", options.tierName),
+		Optional:      !options.required,
+		Computed:      options.tierName == "master",
```

Comment:
Does it make sense to make all tiers besides "hot" `Computed`? Otherwise, users have to explicitly specify all tiers which can be scaled up if autoscaling is enabled. I guess, the original TF issue (which was mentioned in the comment) doesn't prevent us from doing that anymore.
If we decide to do this, we need to make sure that `Required` and `Computed` aren't both true.

Reply:
> the original TF issue (which was mentioned in the comment) doesn't prevent us from doing that anymore.

Not sure if this is the same issue as the comment mentioned, but I did have constant plan changes in node-roles due to a bug in the TF SDK, but I put in a workaround in `node_roles_plan_modifier.go` so it shouldn't happen anymore.

However I don't want to just enable it for all tiers as it isn't necessary for this feature and it may introduce new bugs in certain situations. 

But I think it could be a good change to make in a separate PR. While working on this I noticed also that it could be nice to have the template values already filled in a plan-modifier as well. Then one would see also any tiers and sizes filled in from the template in the plan before applying.